libkbfs/folder_block_ops.go

```diff
@@ -2679,29 +2679,32 @@ func (fbo *folderBlockOps) getDeferredWriteCountForTest(lState *lockState) int {
 func (fbo *folderBlockOps) updatePointer(kmd KeyMetadata, oldPtr BlockPointer, newPtr BlockPointer, shouldPrefetch bool) {
 	// TODO: Remove this comment when we're done debugging because it'll be
 	// everywhere.
-	fbo.log.CDebugf(context.TODO(), "Updating reference for pointer %s to %s", oldPtr.ID, newPtr.ID)
 	updated := fbo.nodeCache.UpdatePointer(oldPtr.Ref(), newPtr)
 	if !updated {
 		return
 	}
-	// Prefetch the new reference, but only if it already exists in the block
-	// cache. Ideally we'd always prefetch it, but we need the type of the
-	// block so that we can call `NewEmpty`.
-	// TODO KBFS-1850: Eventually we should use the codec library's ability to
-	// decode into a nil interface to no longer need to pre-initialize the
-	// correct type.
-	block, err := fbo.config.BlockCache().Get(oldPtr)
-	if err != nil {
-		return
-	}
+	// Only prefetch if the updated pointer is a new block ID.
+	if oldPtr.ID != newPtr.ID {
+		fbo.log.CDebugf(context.TODO(), "Updated reference for pointer %s to %s.", oldPtr.ID, newPtr.ID)
```

Comment:
I tried `DeleteTransient` here, but looks like there are still requests for blocks even after the pointer has been updated.